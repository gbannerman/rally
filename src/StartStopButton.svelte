<script lang="ts">
  import { PlayIcon, PauseIcon } from "svelte-feather-icons";

  export let running: boolean;
  export let onToggle: () => void;
</script>

<div class="StartStopButton">
  <button class={running ? "icon stop" : "icon start"} on:click={onToggle}>
    {#if running}
      <PauseIcon />
    {:else}
      <PlayIcon />
    {/if}
  </button>
</div>

<style>
  .icon {
    padding: 5px;
    margin: 10px;
    height: 40px;
    width: 40px;
    border: 1px transparent solid;
    border-radius: 5px;
    color: #374550;
  }

  .icon.start {
    background-color: #6ac46a;
  }

  .icon.stop {
    background-color: #e5b66c;
  }
</style>
