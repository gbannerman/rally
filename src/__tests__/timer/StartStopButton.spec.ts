import { render, fireEvent } from "@testing-library/svelte";

import StartStopButton from "../../timer/StartStopButton.svelte";

describe("StartStopButton component", () => {
  it("renders correctly", () => {
    const { container } = render(StartStopButton, {
      running: false,
      onToggle: () => {},
    });

    expect(container).toMatchSnapshot();
  });

  it("has correct class when stopped", () => {
    const { getByRole } = render(StartStopButton, {
      running: false,
      onToggle: () => {},
    });

    const button = getByRole("button");

    expect(button).toHaveClass("start");
  });

  it("has correct class when running", () => {
    const { getByRole } = render(StartStopButton, {
      running: true,
      onToggle: () => {},
    });

    const button = getByRole("button");

    expect(button).toHaveClass("stop");
  });

  it("displays play icon when stopped", () => {
    const { getByRole } = render(StartStopButton, {
      running: false,
      onToggle: () => {},
    });

    const button = getByRole("button");
    const icon = button.querySelector("svg");

    expect(icon).toHaveClass("feather-play");
  });

  it("displays pause icon when running", () => {
    const { getByRole } = render(StartStopButton, {
      running: true,
      onToggle: () => {},
    });

    const button = getByRole("button");
    const icon = button.querySelector("svg");

    expect(icon).toHaveClass("feather-pause");
  });

  it("calls onToggle when button clicked", async () => {
    const mockOnToggle = jest.fn();

    const { getByRole } = render(StartStopButton, {
      running: false,
      onToggle: mockOnToggle,
    });

    const button = getByRole("button");

    await fireEvent.click(button);
    expect(mockOnToggle).toHaveBeenCalledTimes(1);
  });
});
