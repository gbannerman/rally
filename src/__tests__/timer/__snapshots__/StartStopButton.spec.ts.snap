// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartStopButton component renders correctly 1`] = `
<body>
  <div>
    <button
      class="icon start svelte-znyfjv"
    >
      <svg
        class="feather feather-play "
        fill="none"
        height="100%"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="5 3 19 12 5 21 5 3"
        />
      </svg>
    </button>
  </div>
</body>
`;
